memcached-operator/pkg/apis/cache/v1alpha1/memcached_types.go

```diff
@@ -24,15 +25,27 @@ type MemcachedStatus struct {
 	// Important: Run "operator-sdk generate k8s" to regenerate code after modifying this file
 	// Add custom validation using kubebuilder tags: https://book-v1.book.kubebuilder.io/beyond_basics/generating_crd.html
 
-	// Status of Nodes
-	Nodes []string `json:"nodes"`
+	// Nodes are the names of the memcached pods
+	// +kubebuilder:validation:UniqueItems=true
+	Nodes PodList `json:"nodes"`
+}
+
+// PodList is a list of names of pods
+type PodList struct {
+	Names []string
+}
+
+func NewPodList(value []string) PodList {
+	p := PodList{Names: value}
+	return p
```

Comment:
I'm not sure I understand the need for nesting this. Is this a structural schema thing? I'd be pretty surprised if structural schema forbids top-level `[]string` values. 

Is there a doc that talks about this?

Reply:
You can easily test it and check the error in the Validations rules. ( just clone the project, add []string and run `operator-sdk generate openapi`).  Following the error:

```
 $ operator-sdk generate openapi
INFO[0000] Running OpenAPI code-generation for Custom Resource group versions: [cache:[v1alpha1], ] 
API rule violation: list_type_missing,github.com/operator-framework/operator-sdk-samples/memcached-operator/pkg/apis/cache/v1alpha1,MemcachedStatus,Nodes
INFO[0006] Created deploy/crds/cache.example.com_memcacheds_crd.yaml 
INFO[0006] Code-generation complete.  
```
Also, see https://github.com/operator-framework/operator-sdk/issues/2114 have a better understanding and check why we did not see it before. 

Also, I working on to try to identify if is an issue in the openapi, if it will be solved in the feature versions and etc .. However .. we cannot keep the example, guide, mock data etc with something that will not work. Also, the Type allow the users to know how they can do that which shows a common question as well.

Reply:
Sounds like maybe we just need an extra annotation that we didn't need before?

Reply:
Let's with who shows be the master on this. @estroz, do you know another way to solve this scenario? Do you think that it can be some annotation?

Reply:
Raised : https://github.com/kubernetes/kube-openapi/issues/175

Reply:
confirmed the []string is no longer acceptable 👍